frontend/src/components/DivisionInputModal.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import Modal from './Modal';
+import { Controller, useForm } from 'react-hook-form';
+import Button from './Button';
+import * as yup from 'yup';
+import i18n from '../common/config/i18n';
+import { yupResolver } from '@hookform/resolvers/yup';
+import Input from './Input';
+import { REGEX } from '../common/constants/patterns';
+
+interface DivisionInputModalProps {
+  title: string;
+  onClose: () => void;
+  onSubmit: (division: DivisionFormTypes) => void;
+  defaultValue?: string;
+}
+
+type DivisionFormTypes = {
+  name: string;
+};
+
+const schema = yup
+  .object({
+    name: yup
+      .string()
+      .required(`${i18n.t('division:form.required')}`)
+      .min(2, `${i18n.t('division:form.min_length')}`)
+      .max(20, `${i18n.t('division:form.max_length')}`)
+      .matches(REGEX.NAME_REGEX, `${i18n.t('division:form.pattern')}`),
+  })
+  .required();
+
+const DivisionInputModal = ({
+  title,
+  onClose,
+  onSubmit,
+  defaultValue,
+}: DivisionInputModalProps) => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<DivisionFormTypes>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+    resolver: yupResolver(schema),
+  });
+
+  return (
+    <Modal title={title} onClose={onClose}>
+      <form className="space-y-6">
+        <Controller
+          key="name"
+          name="name"
+          control={control}
+          render={({ field: { onChange, value } }) => {
+            return (
+              <Input
```

Comment:
If it's part of the form and handles error we should use the FormInput component. Please revalidate.